File: nimbus/View/DetailVC.m
//
//  DetailVC.m
//  nimbus
//
//  Created by Pericles Maravelakis on 19/04/2017.
//	periclesm@cloudfields.net
//	Licensed under Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
//	https://creativecommons.org/licenses/by-sa/4.0/
//

#import "DetailVC.h"
@import SafariServices;

@implementation DetailVC

- (void)viewDidLoad
{
	[self GetCloudData];
    [super viewDidLoad];

	activity.type = DGActivityIndicatorAnimationTypeBallRotate;
	activity.size = activity.frame.size.width-8.0f;
	activity.tintColor = [UIColor lightGrayColor];
}

#pragma mark - Data

- (void)GetCloudData
{	
	CloudList *cl = [DataLogic GetCloudInfo:self.objectId];
	
	self.title = cl.name;
	clInitials.text = cl.initials;
	clName.text = cl.name;
	clAltitude.text = [[DataLogic GetCloudType:cl.type] stringByAppendingString:@" altitude"];
	clDetails.text = [DataLogic GetCloudDetails:cl.detail shortText:NO];

	clImage.alpha = 0;
	[activity startAnimating];
	[Networker GetRemoteImage:[DataLogic GetCloudImageURL:cl.detail] completion:^(UIImage *image) {
		self->clImage.image = image;
		[UIView animateWithDuration:0.25 animations:^{ self->clImage.alpha = 1; }];
		[self->activity stopAnimating];
	}];
}

#pragma mark - IBActions

- (IBAction)WikiButton:(id)sender
{
	CloudList *cl = [DataLogic GetCloudInfo:self.objectId];
	SFSafariViewController *safariView = [[SFSafariViewController alloc] initWithURL:[NSURL URLWithString:[DataLogic GetCloudWikiURL:cl.detail]]];
	[self presentViewController:safariView animated:YES completion:nil];
}

@end

File: nimbus/View/Swift/DetailVC.swift
//
//  DetailVC.swift
//  nimbus
//
//  Created by Pericles Maravelakis on 1/2/20.
//  periclesm@cloudfields.net
//  Licensed under Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)
//  https://creativecommons.org/licenses/by-sa/4.0/
//

import UIKit
import SafariServices

class DetailVC: UITableViewController {
    
    var objectId: String = ""
    
    @IBOutlet weak var clImage: UIImageView!
    @IBOutlet weak var clInitials: UILabel!
    @IBOutlet weak var clName: UILabel!
    @IBOutlet weak var clAltitude: UILabel!
    @IBOutlet weak var clDetails: UILabel!
    @IBOutlet weak var activity: ActivityView!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.getCloudData()
    }
    
    // MARK: - Data
    
    func getCloudData() {
        let cl = DataLogic.getCloudInfo(self.objectId)
        
        self.title = cl?.name
        clInitials.text = cl?.initials
        clName.text = cl?.name
        clAltitude.text = DataLogic.getCloudType(cl?.type) + " altitude"
        clDetails.text = DataLogic.getCloudDetails(cl?.detail, shortText: false)
        
        clImage.alpha = 0
        activity.startAnimating()

		let config = NetConfig.initWithConfig(requestURL: URL(string: DataLogic.getCloudImageURL(cl?.detail)), requestTimeout: 10, requestMethod: .GET)
		Networker.getImage(config: config) { (response) in
			if response.completed {
				self.clImage.image = response.data as? UIImage
				UIView.animate(withDuration: 0.25) {
					self.clImage.alpha = 1
				}
			}

			self.activity.stopAnimating()
		}
        
//        Networker.getRemoteImage(DataLogic.getCloudImageURL(cl?.detail)) { (image) in
//            self.clImage.image = image
//            UIView.animate(withDuration: 0.25) {
//                self.clImage.alpha = 1
//            }
//            self.activity.stopAnimating()
//        }
    }

    // MARK: - Actions
    
    @IBAction func WikiButton(_ sender: UIBarButtonItem) {
		if let cl = DataLogic.getCloudInfo(self.objectId) {
			if let url = URL(string: DataLogic.getCloudWikiURL(cl.detail)) {
				let safari = SFSafariViewController(url: url)
				safari.modalPresentationStyle = .formSheet
				safari.modalTransitionStyle = .coverVertical
				self.present(safari, animated: true, completion: nil)
			}
		}
	}
}
